File: packages/api-models-py/skyportal_py_models/tokens.py
"""Request and response models for ``/api/internal/tokens``."""

from __future__ import annotations

import datetime

from pydantic import BaseModel, ConfigDict


class ApiToken(BaseModel):
    """An API token (upstream baselayer ``Token``).

    The token's ACLs are not serialized; they appear only on the profile's
    token listing (``UserProfile.tokens``).
    """

    model_config = ConfigDict(extra="forbid")

    id: str
    created_at: datetime.datetime | None = None
    modified: datetime.datetime | None = None
    created_by_id: int | None = None
    name: str | None = None


class TokenPostResponse(BaseModel):
    """Result of creating a token."""

    model_config = ConfigDict(extra="forbid")

    token_id: str

File: packages/api-models-ts/src/Tokens.ts
/**
 * Request and response models for `/api/internal/tokens`.
 *
 * @since 1.0.0
 */

import * as v from "valibot";

import * as Schemas from "./Schemas.ts";

/**
 * An API token (upstream baselayer `Token`).
 *
 * The token's ACLs are not serialized; they appear only on the profile's token
 * listing ({@link skyportal-js/Profile!UserProfile}`.tokens`).
 *
 * @since 1.0.0
 * @category Models
 */
export const ApiToken = Schemas.model(
    v.strictObject({
        id: v.string(),
        created_at: Schemas.NullishTimestamp,
        modified: Schemas.NullishTimestamp,
        created_by_id: Schemas.NullishInteger,
        name: Schemas.NullishString,
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type ApiToken = v.InferOutput<typeof ApiToken>;

/**
 * Result of creating a token.
 *
 * @since 1.0.0
 * @category Models
 */
export const TokenPostResponse = Schemas.model(
    v.strictObject({ token_id: v.string() })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type TokenPostResponse = v.InferOutput<typeof TokenPostResponse>;
